### PR TITLE
Make_output function complete

### PR DESCRIPTION
**Pull request by lxbtlr**
looks through matrix and places appropriate tile to output image, assumes default image & tile size, as well as default output image name to be "Output.jpg"

### CODE REVIEW

File: src/imgen.c [hunk 1]
```diff
@@ -13,6 +13,11 @@
 #define STB_IMAGE_WRITE_IMPLEMENTATION
 #include <stb/stb_image_write.h>
 
+// default image width, height
+#define HEIGHT 1080
+#define WIDTH 1080
```
Comment (hnvakil) [inline src/imgen.c:18]: Is this the size of the image displayed in pixels?

File: src/imgen.h [hunk 2]
```diff
@@ -70,4 +70,19 @@ void image_load(Image *img, const char *filename);
...
+ * tile (of entropy 1) to the output image at the equivelant cell coordinates 
+ * in the image. During each cycle of this loop a new temporary image struct is
+ * created, loaded with the appropriate tile, then freed from memory. Once the 
+ * loop completes, all tiles will have been placed; the function saves the image
+ * then frees it from memory.
+ * 
+ * @param cells The (assumed to be) populated matrix. 
+ *
+ */
+void make_output(matrix* cells);
```
Comment (hnvakil) [inline src/imgen.h:88]: Fantastic docstring. Truly enlightening to read, I am now a better man for having read it.

File: src/imgen.c [hunk 3]
```diff
@@ -13,6 +13,11 @@
 #define STB_IMAGE_WRITE_IMPLEMENTATION
 #include <stb/stb_image_write.h>
```
Comment (hnvakil) [inline src/imgen.c:14]: Can we add install instructions for stb_image to the READme?

File: src/helpers.h [hunk 4]
```diff
@@ -5,14 +5,14 @@
...
-//   }\
-// }while(0)
+#define ON_ERROR_EXIT(cond, message) \ 
+do{ \
+  if(cond){ \ 
+    printf("ERROR in function: %s at line %d\n", __func__, __LINE__ ); \
+    perror((message)); \ 
+    exit(1); \ 
+  }\
+}while(0)
```
Comment (hnvakil) [inline src/helpers.h:15]: Is building working for you with the macro defined this way? I can check on my machine in the morning but it cmake was unhappy with me the last time I had the macro defined

**Review by hnvakil — COMMENTED** [review]
Mostly clarifying questions, have we manually tested this at all?

File: src/imgen.c (hunk 1 above)
Comment (lxbtlr) [reply src/imgen.c:18]: Yes

File: src/helpers.h (hunk 4 above)
Comment (lxbtlr) [reply src/helpers.h:15]: We can switch to using a func instead

File: src/imgen.c (hunk 3 above)
Comment (lxbtlr) [reply src/imgen.c:14]: I used a package manager to install it -- but adding it to the default c lib path after downloading from source is effectively the full instructions

Comment (hnvakil) [reply src/imgen.c:14]: Great!

**Review by hnvakil — APPROVED** [review]
Looks good to me with these changes

**Pull request merged**